### PR TITLE
feat: accept fn as parameter in decorators

### PR DESCRIPTION
**Pull request by sheikalthaf**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Currently decorators not accepting the fn as parameter

Issue Number: #31 

## What is the new behavior?
now decorators can accept fn as parameter and it returns string

## Does this PR introduce a breaking change?
```
[ ] Yes
[x ] No
```

## Other information

### CODE REVIEW

File: lib/table-storage/azure-table.interface.ts [hunk 1]
```diff
@@ -38,3 +38,5 @@ export interface Repository<T> {
 
   delete(rowKey: string, entity: T): Promise<AzureTableStorageResponse>;
 }
+
+export type ValueType = ((e: any) => string) | string;
```
Comment (santoshyadav198613) [inline lib/table-storage/azure-table.interface.ts]: Just wondering if (e: Function)  is possible, it will help us avoid passing other values.

Comment (sheikalthaf) [reply lib/table-storage/azure-table.interface.ts]: With the current implemetation we can able to validate the return type as string. If we make it as a `Function` then we can't able to validate the return type

Comment (santoshyadav198613) [reply lib/table-storage/azure-table.interface.ts]: Ok thanks

**Comment by sheikalthaf** [comment]
Any progress in this PR?

**Pull request merged**